lib/systemFlake.nix

```diff
@@ -179,18 +190,24 @@ mergeAny otherArguments (
   eachSystem supportedSystems
     (system:
       let
+        filterAttrs = pred: set:
+          listToAttrs (concatMap (name: let value = set.${name}; in if pred name value then [ ({ inherit name value; }) ] else [ ]) (attrNames set));
+
+        channelFlakes = filterAttrs (_: value: value ? legacyPackages) inputs;
+        channelsFromFlakes = mapAttrs (name: input: { inherit input; }) channelFlakes;
```

Comment:
Not related to this PR, just a thought:
```nix
        channelsFromInputs = _ inputs [
          (filterAttrs (_: value: value ? legacyPackages)) 
          (mapAttrs (name: input: { inherit input; }))
        ];
```
Do you guys think that function like this would get approved as part of nix builtins? I think in lib it is called `pipe` but it would be very nice to have it as builtin function which you don't need to import (like `derivation` or `toString`)

I picked function name `_` because it looks like a language feature and does not conflict  with anything

Reply:
Yeah `pipe` is really useful, could also speed up evaluation if its done in nix.